modules/anyclipBidAdapter.js

```diff
@@ -1,213 +1,222 @@
-    const mediaTypes = deepAccess(bid, 'mediaTypes', {});
-    const banner = deepAccess(mediaTypes, BANNER, {});
-
-    const isValidBidder = (bidder === BIDDER_CODE);
-    const isValidSize = (Boolean(banner.sizes) && isArray(mediaTypes[BANNER].sizes) && mediaTypes[BANNER].sizes.length > 0);
-    const hasSizes = mediaTypes[BANNER] ? isValidSize : false;
-    const hasRequiredBidParams = Boolean(params.publisherId && params.supplyTagId);
-
-    const isValid = isValidBidder && hasSizes && hasRequiredBidParams;
-    if (!isValid) {
-      logError(`Invalid bid request: isValidBidder: ${isValidBidder}, hasSizes: ${hasSizes}, hasRequiredBidParams: ${hasRequiredBidParams}`);
+/**
+ * Make a server request from the list of BidRequests.
+ *
+ * @param {validBidRequest?pbjs_debug=trues[]} - an array of bids
+ * @return ServerRequest Info describing the request to the server.
+ */
+function buildRequests(validBidRequests, bidderRequest) {
+  const {refererInfo = {}, gdprConsent = {}, uspConsent} = bidderRequest;
+  const requests = validBidRequests.map(req => {
+    const request = {};
+    request.bidId = req.bidId;
+    request.banner = deepAccess(req, 'mediaTypes.banner');
+    request.auctionId = req.ortb2?.source?.tid;
+    request.transactionId = req.ortb2Imp?.ext?.tid;
+    request.sizes = parseSizesInput(getAdUnitSizes(req));
+    request.schain = req.schain;
+    request.location = {
+      page: refererInfo.page,
+      location: refererInfo.location,
+      domain: refererInfo.domain,
+      whost: window.location.host,
+      ref: refererInfo.ref,
+      isAmp: refererInfo.isAmp
+    };
+    request.device = {
+      ua: navigator.userAgent,
+      lang: navigator.language
+    };
+    request.env = {
+      publisherId: req.params.publisherId,
+      supplyTagId: req.params.supplyTagId,
+      floor: getBidFloor(req)
+    };
+    request.ortb2 = req.ortb2;
+    request.ortb2Imp = req.ortb2Imp;
+    request.tz = new Date().getTimezoneOffset();
+    request.ext = req.params.ext;
+    request.bc = req.bidRequestsCount;
+
+    if (req.userIdAsEids && req.userIdAsEids.length !== 0) {
+      request.userEids = req.userIdAsEids;
+    } else {
+      request.userEids = [];
+    }
+    if (gdprConsent.gdprApplies) {
+      request.gdprApplies = Number(gdprConsent.gdprApplies);
+      request.consentString = gdprConsent.consentString;
+    } else {
+      request.gdprApplies = 0;
+      request.consentString = '';
+    }
+    if (uspConsent) {
+      request.usPrivacy = uspConsent;
+    } else {
+      request.usPrivacy = '';
     }
-    return isValid;
-  },
-
-  /**
-   * @param {BidRequest[]} validBidRequests
-   * @param {*} bidderRequest
-   * @return {ServerRequest}
-   */
-  buildRequests: (validBidRequests, bidderRequest) => {
-    const bidRequest = validBidRequests[0];
-
-    let refererInfo;
-    if (bidderRequest && bidderRequest.refererInfo) {
-      refererInfo = bidderRequest.refererInfo;
+    if (config.getConfig('coppa')) {
```

Comment:
Can you please access coppa using `bidderRequest.ortb2.regs.coppa` instead of through config.

Reply:
Hello @ChrisHuie,
Thank you for the review. We have decided to remove this block, as we have the full ORTB2 object in the request. We will proceed with this value on the adserver.
Let me know if you need any further adjustments!